orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+        *priority = 0;
+        return NULL;
+    }
+
+    if (init_done) {
+        *priority = 2;
+        return &orte_rml_ofi.super;
+    }
+
+    *priority = 2;
+
+
+    /**
+     * Hints to filter providers
+     * See man fi_getinfo for a list of all filters
+     * mode:  Select capabilities MTL is prepared to support.
+     *        In this case, MTL will pass in context into communication calls
+     * ep_type:  reliable datagram operation
+     * caps:     Capabilities required from the provider.
+     *           Tag matching is specified to implement MPI semantics.
+     * msg_order: Guarantee that messages with same tag are ordered.
+     */
+
+    hints = fi_allocinfo();
+    if (!hints) {
+        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                            "%s:%d: Could not allocate fi_info\n",
+                            __FILE__, __LINE__);
+        return NULL;
+    }
+
+    /**
+     * Refine filter for additional capabilities
+     * endpoint type : Reliable datagram
+     * threading:  Disable locking
+     * control_progress:  enable async progress
+     */
+     hints->mode               = FI_CONTEXT;
+     hints->ep_attr->type      = FI_EP_RDM;      /* Reliable datagram         */
+
+     hints->domain_attr->threading        = FI_THREAD_UNSPEC;
+     hints->domain_attr->control_progress = FI_PROGRESS_AUTO; 
+     hints->domain_attr->data_progress = FI_PROGRESS_AUTO; 
+     hints->domain_attr->av_type          = FI_AV_MAP;
+
+    /**
+     * FI_VERSION provides binary backward and forward compatibility support
+     * Specify the version of OFI is coded to, the provider will select struct
+     * layouts that are compatible with this version.
+     */
+     fi_version = FI_VERSION(1, 0);
+
+    /**
+     * fi_getinfo:  returns information about fabric  services for reaching a
+     * remote node or service.  this does not necessarily allocate resources.
+     * Pass NULL for name/service because we want a list of providers supported.
+     */
+     ret = fi_getinfo(fi_version,    /* OFI version requested                    */
+                       NULL,          /* Optional name or fabric to resolve       */
+                       NULL,          /* Optional service name or port to request */
+                       0ULL,          /* Optional flag                            */
+                       hints,        /* In: Hints to filter providers            */
+                       &orte_rml_ofi.fi_info_list);   /* Out: List of matching providers          */
+     if (0 != ret) {
+        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                            "%s:%d: fi_getinfo failed: %s\n",
+                            __FILE__, __LINE__, fi_strerror(-ret));
+    } else {
+
+        /* added for debug purpose - Print the provider info
+        print_transports_query();
+        print_provider_list_info(orte_rml_ofi.fi_info_list);
+        */
```

Comment:
If these calls are commented out, will the compiler complain about orphaned functions?

Instead, should these functions be called when verbosity is set high enough?